bindings_swift/src/types.rs

```diff
@@ -93,12 +93,13 @@ pub struct QueryResponse {
 }
 
 impl QueryResponse {
-    pub fn envelopes(self) -> Vec<crate::Envelope> {
-        self._envelopes
+    pub fn envelopes(&self) -> Vec<crate::Envelope> {
+        self._envelopes.clone()
```

Comment:
I'm curious about the intended ownership here. 

Who owns the QueryResponse? is passing a ref to the Vec not an option?

Want to make sure were only cloning when needed

Reply:
It's a good question, AFAICT the issue here boils down to how `swift-bridge` emulates ownership on the Swift side.

See this code: https://github.com/xmtp/xmtp-rust-swift/blob/main/Sources/XMTPRust/xmtp_rust_swift.swift#L406 which looks like:
```
extension QueryResponse {
    public func envelopes() -> RustVec<Envelope> {
        RustVec(ptr: __swift_bridge__$QueryResponse$envelopes({isOwned = false; return ptr;}()))
    }

    public func paging_info() -> Optional<PagingInfo> {
        __swift_bridge__$QueryResponse$paging_info({isOwned = false; return ptr;}()).intoSwiftRepr()
    }
}
```

It appears that on either call, ownership is "given up" via an explicitly tracked variable which tells Swift it's okay to free the memory on `deinit`. I don't pretend to fully understand why the `deinit` is invoked while the `response` is still within scope on the Swift side.